app/Database/Requirement.hs

```diff
@@ -0,0 +1,59 @@
+{-|
+Description: Requirement
+
+Module containing data type that represents a "Requirement".
+
+We will use parsed data to create instances of this type.
+-}
+module Requirement
+( ProgramReq(..)
+, CourseRequirement(..)
+, CourseReq(..)
+, Req(..)
+, showProgramReq
+, showCourseRequirement
+, showCourseReq
+, showReq
+) where
+
+import Data.String
+
+
+data ProgramReq = PRGREQ String [Req]
+
+-- | Returns a well formatted String representing a program requirement for specified program.
+showProgramReq :: ProgramReq -> String
+showProgramReq (PRGREQ program reqs) = "The program requirements for " + program + ":\n"
+										+ map (showReq) reqs
+
+
+-- Assuming we have correctly parsed inputs, CourseReq is in order [corequisite, exclusion, prerequisite]
+data CourseRequirement = CRSREQ String [CourseReq]
+
+-- | Returns a well formatted String representing all course requirements for specified course.
+showCourseRequirement :: CourseRequirement -> String
+showCourseRequirement (CRSREQ course coursereqs) = course + "\n" + map (showCourseReq) coursereqs 
+
+
+data CourseReq = CREQ String Req | EXCL String Req | PREQ String Req
```

Comment:
You don't need this type. `CourseRequirement` can have type `CRSREQ String Req Req Req`, where the first `Req` is the corequisite, the second is the exclusion, and the third is the prerequisite.